src/widgets/calendar/lv_calendar_header_dropdown.c

```diff
@@ -121,8 +135,17 @@ static void year_event_cb(lv_event_t * e)
 
     const lv_calendar_date_t * d;
     d = lv_calendar_get_showed_date(calendar);
+
+    /* Get the first year on the options list
+     * NOTE: Assumes the first 4 digits in the option list are numbers */
+    char year_buf[YEAR_STR_BUFFER_LEN] = {0};
+    const char * year_p = lv_dropdown_get_options(dropdown);
+    strncpy(year_buf, year_p, DIGITS_IN_YEAR);
+    const uint32_t year = (year_buf[0] - '0') * 1000 + (year_buf[1] - '0') * 100 + (year_buf[2] - '0') * 10 +
+                          (year_buf[3] - '0');
+
```

Comment:
Is it unnecessary to use the intermediate variable `year_buf`?
```c
const char * year_p = lv_dropdown_get_options(year_dd);
const uint32_t year = (year_p[0] - '0') * 1000 + (year_p[1] - '0') * 100 + (year_p[2] - '0') * 10 +
                          (year_p[3] - '0');
```

Reply:
I think so, I will update the code accordingly.